tests/sles4sap/sap_cluster.pm

```diff
@@ -0,0 +1,162 @@
+    if (check_var('CLUSTER_NAME', 'hana')) {
+        my $automated_register = get_required_var('AUTOMATED_REGISTER');
+        my ($virtual_ip, $virtual_netmask) = split '/', get_required_var('VIRTUAL_IP_CIDR');
+        push @sedoptions, "-e 's|%AUTOMATED_REGISTER%|$automated_register|g'";
+        push @sedoptions, "-e 's|%VIRTUAL_IP_ADDRESS%|$virtual_ip|g'";
+        push @sedoptions, "-e 's|%VIRTUAL_IP_NETMASK%|$virtual_netmask|g'";
+    }
+
+    # We need to execute the sed command node by node
+    my $cmd          = undef;
+    my $cluster_conf = undef;
+    if (get_var('NW')) {
+        $cluster_conf = 'nw_cluster.conf';
+    } else {
+        $cluster_conf = 'hana_cluster.conf';
+    }
+    assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/$cluster_conf -o /tmp/$cluster_conf";
+    $cluster_conf = "/tmp/" . $cluster_conf;
+
+    # Avoid hostname change on reboot.  See bsc#1153402
+    assert_script_run "sed -i 's/DHCLIENT_SET_HOSTNAME=\"yes\"/DHCLIENT_SET_HOSTNAME=\"no\"/' /etc/sysconfig/network/dhcp";
+
+    my $node1 = choose_node(1);
+    my $node2 = choose_node(2);
+
+    if (is_node(1)) {
+        # Initiate the template
+        $cmd = 'sed -i' . join(' ', @sedoptions) . " $cluster_conf";
+        assert_script_run $cmd;
+
+        if (get_var('HANA')) {
+            foreach ($node1, $node2) {
+                add_to_known_hosts($_);
+            }
+            assert_script_run "scp -qr /usr/sap/$sid/SYS/global/security/rsecssfs/* root\@$node2:/usr/sap/$sid/SYS/global/security/rsecssfs/";
+            my $password = get_required_var('PASSWORD');
+            assert_script_run qq(su - $sapadm -c "hdbsql -u system -p $password -i $instance_id -d SYSTEMDB \\"BACKUP DATA FOR FULL SYSTEM USING FILE ('backup')\\""), 300;
+            assert_script_run "su - $sapadm -c 'hdbnsutil -sr_enable --name=NODE1'";
+        }
+
+        # Synchronize the nodes
+        barrier_wait "SAP_INIT_CONF_$cluster_name";
+        barrier_wait "SAP_CREATED_CONF_$cluster_name";
+
+        # Upload the configuration into the cluster
+        assert_script_run 'crm configure property maintenance-mode=true';
+        assert_script_run "crm configure load update $cluster_conf";
+        assert_script_run 'crm configure property maintenance-mode=false';
+    }
+    else {
+        # Synchronize the nodes
+        barrier_wait "SAP_INIT_CONF_$cluster_name";
+
+        if (get_var('HANA')) {
+            assert_script_run "su - $sapadm -c 'sapcontrol -nr $instance_id -function StopSystem HDB'";
+            sleep 10;
+            assert_script_run "su - $sapadm -c 'hdbnsutil -sr_register --remoteHost=$node1 -remoteInstance=$instance_id --replicationMode=sync --name=NODE2 --operationMode=logreplay'";
+            assert_script_run "su - $sapadm -c 'sapcontrol -nr $instance_id -function StartSystem HDB'";
+        }
+
+        # Use mutex to be sure that only *one* node at a time can access the file
+        mutex_lock 'support_server_ready';
```

Comment:
Is this really needed? Files are locally in `/tmp/$cluster_conf` in each node, so I don't think there's any risk of more than one process writing on it at the time. Unless this runs on the support server, but it doesn't seem to be the case.

Reply:
> Is this really needed? Files are locally in `/tmp/$cluster_conf` in each node, so I don't think there's any risk of more than one process writing on it at the time. Unless this runs on the support server, but it doesn't seem to be the case.

Agree.  We're also modifying the template in the secondary node when there's no need for that.